test/mysql2_test.rb

```diff
@@ -312,32 +314,35 @@ def client.raw_ping
   def test_changes_timeout_when_half_open_and_configured
     client = connect_to_mysql!(half_open_resource_timeout: 1)
 
-    @proxy.downstream(:latency, latency: 3000).apply do
-      (ERROR_THRESHOLD * 2).times do
-        assert_raises Mysql2::Error do
-          client.query('SELECT 1 + 1;')
+    Timecop.freeze do
+      @proxy.downstream(:latency, latency: 3000).apply do
+        ERROR_THRESHOLD.times do
+          assert_raises Mysql2::Error do
+            client.query('SELECT 1 + 1;')
+          end
         end
       end
-    end
 
-    assert_raises Mysql2::CircuitOpenError do
-      client.query('SELECT 1 + 1;')
+      assert_raises Mysql2::CircuitOpenError do
+        client.query('SELECT 1 + 1;')
+      end
     end
 
-    Timecop.travel(ERROR_TIMEOUT + 1) do
+    time_circuit_half_open = ERROR_TIMEOUT + 1
+    Timecop.travel(time_circuit_half_open) do
       @proxy.downstream(:latency, latency: 1500).apply do
         assert_raises Mysql2::Error do
           client.query('SELECT 1 + 1;')
         end
       end
```

Comment:
I updated the mysql2 test to more closely match the Redis test. However, in doing so I noticed some odd behaviour. 

I tried to replace the above four lines with a call `assert_mysql_timeout_in_delta` for a timeout of 1s. The timeout raises successfully, but for some reason it takes `4.6 seconds` to timeout, instead of 1s.

I also tried the same thing, but with a lower timeout, to see if the circuit was just in a bad state, and the timeout doesn't occur. This indicates to me that the timeout is indeed `1 second` but for some reason it takes `mysql2` 4.6 seconds to actually fail. 

This is some odd behaviour and is outside the scope of this PR to fix. It would be good to understand why this happens anyway.